src/ILCompiler.Compiler/src/Compiler/MetadataGeneration.cs

```diff
@@ -133,6 +137,9 @@ private void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
                     return;
                 }
 
+                if (method.HasInstantiation && !method.IsAbstract)
```

Comment:
It's not clear to me why we have the `if` here when `IsMethodEligibleForTracking` will be called anyway to filer out this (and more) anyway.